service/src/main/kotlin/fi/espoo/evaka/invoicing/service/DraftInvoiceGenerator.kt

```diff
@@ -113,7 +112,13 @@ class DraftInvoiceGenerator(
                         data.areaIds,
                         data.businessDays,
                         data.feeThresholds,
-                        absencesByChild,
+                        getChildAbsences =
+                            memoize { child: ChildId ->
+                                (absencesByChild[child] ?: emptyList()).associateBy(
```

Comment:
I think there can be up to 2 absence types per date so `associate` kind of randomly looses one? Not sure if the same issue exists in the original code too.

Reply:
Only billable absences are fetched so there can be only one per date. It's just not obvious from the type `List<AbsenceStub>`, which is one of the reasons why I've changed most functions to accept absence data using types that either make this either explicit (e.g. `Map<LocalDate, AbsenceType>`) or remove the need for this detail completely (e.g. `hasPlannedAbsence: (date: LocalDate) => Boolean`)

Reply:
ah gotcha, all good then